procedures: mounting secrets as a file or as an environment variable

max-cx opened the pull request:
> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

### What does this PR do?

***PR 2062 must be merged before this!***

Moved the content of two example files into two corresponding procedure files because the example files contained some procedure steps. Rewrote the procedure files on the basis of the added content from the example files and then deleted the example files.

Previews:

https://610cebff2175637ba96a0b0a--eclipse-che-docs-pr.netlify.app/che-7/end-user-guide/mounting-a-secret-as-a-file-or-an-environment-variable-into-a-workspace-container/#mounting-a-secret-as-a-file-into-a-workspace-container_che

https://610cebff2175637ba96a0b0a--eclipse-che-docs-pr.netlify.app/che-7/end-user-guide/mounting-a-secret-as-a-file-or-an-environment-variable-into-a-workspace-container/#mounting-a-secret-as-an-environment-variable-into-a-workspace-container_che

### What issues does this PR fix or reference?
3166

### Specify the version of the product this PR applies to.
7.32, 7.34

### PR Checklist

As the author of this Pull Request I made sure that:

- [x] `vale` has been run successfully against the PR branch
- [x] Link checker has been run successfully against the PR branch
- [ ] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
- [ ] Changed article references are updated where they are used (or a redirect has been configured on the docs side):
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)

max-cx committed:
first draft

max-cx committed:
rewrote two example files

max-cx committed:
replaced comma with brackets and reworded

max-cx committed:
Update modules/end-user-guide/examples/che-ref_a-kubernetes-secret-as-a-file.adoc

Co-authored-by: Rolfe Dlugy-Hegwer <rolfedh@users.noreply.github.com>

max-cx committed:
Update modules/end-user-guide/examples/che-ref_a-kubernetes-secret-as-a-file.adoc

Co-authored-by: Rolfe Dlugy-Hegwer <rolfedh@users.noreply.github.com>

max-cx committed:
Update modules/end-user-guide/examples/che-ref_a-kubernetes-secret-as-a-file.adoc

Co-authored-by: Rolfe Dlugy-Hegwer <rolfedh@users.noreply.github.com>

max-cx committed:
Update modules/end-user-guide/examples/che-ref_a-kubernetes-secret-as-a-file.adoc

Co-authored-by: Rolfe Dlugy-Hegwer <rolfedh@users.noreply.github.com>

max-cx committed:
rewrote a few sentences

max-cx committed:
forgot an *

max-cx committed:
fixed one sentence, added a comment line

max-cx committed:
fixed a few lines

max-cx committed:
Update che-ref_a-kubernetes-secret-as-a-file.adoc

max-cx committed:
Update che-ref_a-kubernetes-secret-as-an-variable.adoc

max-cx committed:
Update che-ref_a-kubernetes-secret-as-a-file.adoc

max-cx committed:
work in progress on the final NOTE

max-cx committed:
Update modules/end-user-guide/examples/che-ref_a-kubernetes-secret-as-a-file.adoc

Co-authored-by: Rolfe Dlugy-Hegwer <rolfedh@users.noreply.github.com>

MichalMaler committed:
Merge branch 'master' into 2885max-cx

max-cx committed:
Update che-ref_a-kubernetes-secret-as-a-file.adoc

max-cx committed:
Update che-ref_a-kubernetes-secret-as-a-file.adoc

max-cx committed:
Update che-ref_a-kubernetes-secret-as-a-file.adoc

max-cx committed:
Update che-ref_a-kubernetes-secret-as-a-file.adoc

max-cx committed:
Update modules/end-user-guide/examples/che-ref_a-kubernetes-secret-as-an-variable.adoc

Co-authored-by: Rolfe Dlugy-Hegwer <rolfedh@users.noreply.github.com>

max-cx committed:
Update che-ref_a-kubernetes-secret-as-a-file.adoc

MichalMaler committed:
Merge branch 'master' into 2885max-cx

max-cx committed:
Rewrote one paragraph

max-cx committed:
test

max-cx committed:
added a missing `data` to `data key`

max-cx committed:
minor fixes

max-cx committed:
minor fix

max-cx committed:
reordered sentences and preview test

max-cx committed:
run-on ul item

max-cx committed:
added missing apostrophes

max-cx committed:
adding missing apostrophes

max-cx committed:
Merge branch 'master' into 2885max-cx

MichalMaler committed:
Merge branch 'master' into 2885max-cx

MichalMaler committed:
Merge branch 'master' into 2885max-cx

MichalMaler committed:
Merge branch 'master' into 2885max-cx

max-cx committed:
Update modules/end-user-guide/examples/che-ref_a-kubernetes-secret-as-a-file.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/end-user-guide/examples/che-ref_a-kubernetes-secret-as-a-file.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/end-user-guide/examples/che-ref_a-kubernetes-secret-as-a-file.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/end-user-guide/examples/che-ref_a-kubernetes-secret-as-an-variable.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
removed one comment line

MichalMaler committed:
Merge branch 'master' into 2885max-cx

max-cx committed:
removed a comment line

max-cx committed:
removed a comment line

max-cx committed:
draft

max-cx committed:
draft

max-cx committed:
[3166] Incorporated examples into procedures

The contents of the following example files
modules/end-user-guide/examples/che-ref_a-kubernetes-secret-as-a-file.adoc
modules/end-user-guide/examples/che-ref_a-kubernetes-secret-as-an-variable.adoc

have been incorporated into the following procedure files
modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc
modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc

The two example files have been deleted.
The two procedure files have been rewritten.

Bug: https://issues.redhat.com/browse/RHDEVDOCS-3166

max-cx committed:
Merge branch 'master' into 3166

max-cx committed:
tried fixing ref's

max-cx committed:
Merge branch 'master' into 3166

rolfedh commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,82 @@ WARNING: On
 {platforms-name} older than v1.13 and
 endif::[]
-OpenShift {ocp3-ver}, secrets mounted as file overrides volume mounts defined in the devfile.
+OpenShift {ocp3-ver}, secrets mounted as a file override volume mounts defined in the devfile.
```
Conditionalization and attributes are very susceptible to minor errors and oversights. Please verify the output to make sure it does what you expect. Here, it seems like there might be an issue with "On Kubernetes or OpenShift older than v1.13 and OpenShift 3.11,..."

max-cx replied:
@rolfedh, many thanks for catching this. I started checking the output when I discovered the docs previews.

rolfedh commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,82 @@ WARNING: On
+OpenShift {ocp3-ver}, secrets mounted as a file override volume mounts defined in the devfile.
 
-This section describes how to mount a secret from the user's {orch-namespace} as a file in single-workspace or multiple-workspace containers of {prod-short}.
+A secret can be mounted from the user's {orch-namespace} as a file in single-workspace or multiple-workspace containers of {prod-short}.
```
```suggestion
As a <name of user's role - e.g., developer>, you can mount a secret from your {orch-namespace} as a file in single-workspace or multiple-workspace containers of {prod-short}.
```
- " of {prod-short}." sounds odd. Is there a more common way or saying the same thing or can it be removed?
- Can we address the user more directly? IBM SG states:
```
Second person
As much as possible, use the second person, which is indicated by the personal pronoun you. The second person focuses on the reader.
> When you create a database, you must provide a unique name.

To instruct the user to perform an action, use the imperative mood. The subject of an imperative sentence is understood to be you.
> To begin processing, press Enter.
```

rolfedh replied:
In other words, avoid using "the user" to address the user.

rolfedh replied:
On the other hand, if the audience is an administrator who is configuring the environment for users, it might make sense to talk about "the user" in the third person.  It can help to name the audience in the opening phrase: "As an administrator, you can..."

max-cx replied:
@rolfedh I've rephrased this as follows: "In {prod-short}, you can mount a secret from your {orch-namespace} as a file in single-workspace or multiple-workspace containers."

rolfedh commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,82 @@ WARNING: On
-* A running instance of {prod-short}. To install an instance of {prod-short}, see {link-installing-an-instance}.
+* A running instance of {prod-short}.
++
+To install an instance of {prod-short}, see {link-installing-an-instance}.
```
```suggestion
```

max-cx replied:
Already fixed this according to our other conversation.

rolfedh commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,82 @@ WARNING: On
 .Prerequisites
 
-* A running instance of {prod-short}. To install an instance of {prod-short}, see {link-installing-an-instance}.
+* A running instance of {prod-short}.
```
```suggestion
* A running instance of {prod-short}. See {link-installing-an-instance}.
```

rolfedh replied:
Can we combine these two phrases or remove the cross-reference altogether?

MichalMaler replied:
That combination is good.

max-cx replied:
Fixed this in both files.

rolfedh commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,82 @@ WARNING: On
 .Procedure
 
-. Create a new {platforms-name} secret in the {platforms-name} {orch-namespace} where a {prod-short} workspace will be created.
+To mount a secret as a file, complete the following steps as shown in the next example:
```
```suggestion
```
Can we remove this phrase?  RH SSG https://redhat-documentation.github.io/supplementary-style-guide/#lead-in-sentences states "Do not use a lead-in sentence in the Prerequisites or Procedure sections of a module unless it is necessary to aid navigation or add clarity."

max-cx replied:
Removed. Thank you.

rolfedh commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,82 @@ WARNING: On
+To mount a secret as a file, complete the following steps as shown in the next example:
+
+. Create a new {platforms-name} secret in the {platforms-namespace} of the {prod-short} workspace.
+. Match the secret's `labels` to the set of labels configured in the `che.workspace.provision.secret.labels` property of {prod-short}.
```
```suggestion
. Match the `labels` in the secret with the labels in the `che.workspace.provision.secret.labels` property.
```
- Can we simplify this phrase and improve its clarity, or does this suggestion remove too much information? 
- Also, is it obvious how to match labels in the software?

max-cx replied:
Well, here it doesn't seem to be the case that the `labels` are **in** the secret (these labels identify the secret wherever they appear), so "Match the `labels` in the secret" is likely not going to work here. "Match the `labels` of/for the secret ..." could work, but at that point there would be little distinction between "secret's labels" and "labels of the secret", and the latter would also be longer and make it harder to follow the whole sentence.

rolfedh commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,82 @@ WARNING: On
+** `'false'` disables automatic mounting into all workspace containers. To mount a secret, a user must request it in the devfile component by using the `automountWorkspaceSecrets:true` property.
+. Set a `mount-path` for the file.
+. Include the `mount-as: file` annotation.
+. Add the file's name in the secret's `data` field.
```
```suggestion
. Add the file name in the `data` field of the secret.
```

max-cx replied:
Fixed. Thanks.

rolfedh commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,82 @@ WARNING: On
+. Include the `mount-as: file` annotation.
+. Add the file's name in the secret's `data` field.
+
+NOTE: A {orch-name} secret can contain several `data` items. The names of `data` items must match the name of the secret's file mounted into the container.
```
Does `{orch-name}` render as OpenShift sometimes? When that happens, the indefinite article would need to become "An." It might be easier to replace "A" with "The" or remove `{orch-name}` altogether if it is obvious from the context.

max-cx replied:
@rolfedh Great catch! Thank you, Rolfe! Fixed.

rolfedh commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,82 @@ WARNING: On
+----
+<1> Default label.
+<2> Default label.
+<3> Enabled automatic mounting into all workspace containers.
```
```suggestion
<3> Enables automatic mounting into all workspace containers.
```

max-cx replied:
Fixed. Thanks.

rolfedh commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,82 @@ WARNING: On
+<1> Default label.
+<2> Default label.
+<3> Enabled automatic mounting into all workspace containers.
+<4> Mount path, `{prod-home}/.m2/`, set for the file in all workspace containers.
```
```suggestion
<4> Sets the mount path of the file for all workspace containers.
```

max-cx replied:
Fixed. Thanks.

rolfedh commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,82 @@ WARNING: On
+<2> Default label.
+<3> Enabled automatic mounting into all workspace containers.
+<4> Mount path, `{prod-home}/.m2/`, set for the file in all workspace containers.
+<5> Included `mount-as: file` annotation.
```
```suggestion
<5> Mounts the secret as a file. 
```
Are the other options described in a separate reference document?

max-cx replied:
@rolfedh Actually, all of these are described in the procedure steps that precede the example, and that's why I phrased the callouts this way. Also, I tried to make the structure of the callouts more or less parallel. This particular callout is phrased because of step 5 ("Include the `mount-as: file` annotation") in the procedure, which in the example becomes "Included `mount-as: file` annotation."

max-cx replied:
Fixed. Thanks.

rolfedh commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,82 @@ WARNING: On
+<3> Enabled automatic mounting into all workspace containers.
+<4> Mount path, `{prod-home}/.m2/`, set for the file in all workspace containers.
+<5> Included `mount-as: file` annotation.
+<6> Name of the file to mount, `settings.xml`.
```
```suggestion
<6> The name and encoding of the file to mount.
```
For callouts, it is probably better to provide an explanation without repeating the content of the line being called out.

max-cx replied:
@rolfedh My reasoning for quoting line content: (a) to more clearly link the content of the example with the procedure steps, and (b) to emphasize a specific string on the marked line.

rolfedh commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,82 @@ WARNING: On
+<5> Included `mount-as: file` annotation.
+<6> Name of the file to mount, `settings.xml`.
+
+.Procedure
```
Use a separate module for each procedure.

max-cx replied:
OK, will do.

rolfedh commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,82 @@ WARNING: On
 
-* `app.kubernetes.io/part-of: che.eclipse.org` 
-* `app.kubernetes.io/component: workspace-secret`:
+Optionally, to override the mount path of a secret for a workspace component, edit the devfile as follows:
```
Help your audience understand *why* they would choose one procedure over the other.

rolfedh commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc:
```diff
@@ -5,19 +5,81 @@
 = Mounting a secret as an environment variable into a workspace container
 
-The following section describes how to mount a {platforms-name} secret from the user’s {orch-namespace} as an environment variable, or variables, into single-workspace or multiple-workspace containers of {prod-short}.
+A secret can be mounted from the user’s {orch-namespace} as an environment variable (or variables) in single-workspace or multiple-workspace containers of {prod-short}.
```
Extend the suggestions from the previous topic to this one.

max-cx replied:
@rolfedh Thank you, Rolfe!

max-cx committed:
Merge branch 'master' into 3166

max-cx committed:
[3166] Split a procedure into a separate proc_

Create a separate file and moved the procedure into it --
proc_overriding-a-secrets-mount-path-for-a-workspace-component.adoc
that previously was in
proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc

Bug: https://issues.redhat.com/browse/RHDEVDOCS-3166

max-cx committed:
minor fixes

max-cx committed:
clarified mentions of default labels

max-cx committed:
rewrote two callouts

max-cx committed:
rewrote one callout

max-cx committed:
'a {platforms-name}' > 'the {platforms-name}'

'a' doesn't work here as 'a OpenShift'

max-cx committed:
add 'the {platforms-name}' before 'secret

max-cx committed:
add 'the {platforms-name}' before 'secret'

max-cx committed:
[3166] rewrite step 1 of procedure

rewrote the procedure in both files due to the attributes

 Changes to be committed:
	modified:   modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc
	modified:   modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc

Bug: https://issues.redhat.com/browse/RHDEVDOCS-3166

max-cx committed:
Merge branch 'master' into 3166

max-cx committed:
remove attribute due to indefinite article complication and local redundancy

max-cx committed:
replace unordered list by description list

max-cx committed:
fix one sentence

max-cx committed:
fix description list rendering issue

themr0c commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,55 @@ WARNING: On
 ifeval::["{project-context}" == "che"]
 {platforms-name} older than v1.13 and
```
```suggestion
{kubernetes} older than v1.13 and
```

themr0c commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -9,21 +9,55 @@ WARNING: On
+. Set a `mount-path` for the file.
+. Include the `mount-as: file` annotation.
+. Set the filename in the secret's `data` field.
+
```
```suggestion
+
```

themr0c commented on modules/end-user-guide/partials/proc_overriding-a-secrets-mount-path-for-a-workspace-component.adoc:
```diff
@@ -0,0 +1,42 @@
+
+WARNING: On 
+ifeval::["{project-context}" == "che"]
+{platforms-name} older than v1.13 and
```
```suggestion
{kubernetes} older than v1.13 and
```

themr0c commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc:
```diff
@@ -1,23 +1,80 @@
+<1> Environment variable provisioned with the `myvalue` value.
+<2> Environment variable provisioned with the `othervalue` value.
+<3> Key-value pair provisioned with the `FOO_ENV` environment variable.
+<4> Key-value pair provisioned with the `OTHER_ENV` environment variable.
```
```suggestion
.Example: Mounting a secret as an environment variable
[source,yaml,subs="+quotes,+attributes,+macros"]
----
apiVersion: v1
kind: Secret
metadata:
  name: mvn-settings-secret
  annotations:
    che.eclipse.org/automount-workspace-secret: 'true' <1>
    che.eclipse.org/mount-as: env <2>
    che.eclipse.org/__<mykey>__env-name: __<FOO_ENV>__ <3>
    che.eclipse.org/__<otherkey>___env-name: __<OTHER_ENV>__ <4>
  labels:
    app.kubernetes.io/part-of: che.eclipse.org <5>
    app.kubernetes.io/component: workspace-secret <6>
data:
  __<mykey>__: __<myvalue>__ <7>
  __<otherkey>__: __<othervalue>__ <8>
----
<1> Enables automatic mounting into all workspace containers.
<2> Mounts the secret as a variable. 
<3> Provisions the `__<FOO_ENV>__` environment variable into all workspace containers using the value of __<mykey>__.
<4> Provisions the `__<OTHER_ENV>__` environment variable into all workspace containers using the value of __<otherkey>__.
<5> Matches the configured default label.
<6> Matches the configured default label.
<7> Provisions the `__<myvalue>__` data key value into all workspace containers.
<8> Provisions the `__<othervalue>__` data key value into all workspace containers.
```

max-cx committed:
Update modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc

Co-authored-by: Fabrice Flore-Thébault <ffloreth@redhat.com>

max-cx committed:
Update modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc

Co-authored-by: Fabrice Flore-Thébault <ffloreth@redhat.com>

max-cx committed:
Update modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc

Co-authored-by: Fabrice Flore-Thébault <ffloreth@redhat.com>

max-cx committed:
fix: replaced user-replaced values

- to prevent confusion and make logical connections more obvious

max-cx committed:
Merge branch 'master' into 3166

max-cx committed:
fix: singular > plural

max-cx committed:
fix: added brackets to one sentence

max-cx committed:
fix: added brackets to one sentence

themr0c committed:
Update modules/end-user-guide/partials/proc_overriding-a-secrets-mount-path-for-a-workspace-component.adoc

themr0c committed:
Merge branch 'master' into 3166

themr0c committed:
Merge branch 'master' into 3166

themr0c committed:
Merge branch 'master' into 3166

rkratky commented on modules/end-user-guide/partials/assembly_mounting-secrets-as-file-or-env-variable-in-workspace-container.adoc:
```diff
@@ -11,25 +11,27 @@ Secrets are {platforms-name} objects that store sensitive data such as user name
+A secret can be mounted into a workspace container as a file or an environment variable:
 
-* A file - This creates automatically mounted Maven settings that will be applied to every new workspace with Maven capabilities.
+A file:: Creates automatically mounted Maven settings that will be applied to every new workspace with Maven capabilities.
```
```suggestion
File:: Creates automatically mounted Maven settings that will be applied to every new workspace with Maven capabilities.
```

rkratky commented on modules/end-user-guide/partials/assembly_mounting-secrets-as-file-or-env-variable-in-workspace-container.adoc:
```diff
@@ -11,25 +11,27 @@ Secrets are {platforms-name} objects that store sensitive data such as user name
+A file:: Creates automatically mounted Maven settings that will be applied to every new workspace with Maven capabilities.
 
-* An environment variable - This uses SSH key pairs and AWS authorization tokens for automatic authentication. 
+An environment variable:: Uses SSH key pairs and AWS authorization tokens for automatic authentication. 
```
```suggestion
Environment variable:: Uses SSH key pairs and AWS authorization tokens for automatic authentication. 
```

rkratky commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -7,23 +7,57 @@
+{kubernetes} older than v1.13 and
 endif::[]
-OpenShift {ocp3-ver}, secrets mounted as file overrides volume mounts defined in the devfile.
+OpenShift {ocp3-ver}, secrets mounted as a file override volume mounts defined in the devfile.
```
```suggestion
OpenShift {ocp3-ver}, secrets mounted as files override volume mounts defined in the devfile.
```

rkratky commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -7,23 +7,57 @@
+OpenShift {ocp3-ver}, secrets mounted as a file override volume mounts defined in the devfile.
 
-This section describes how to mount a secret from the user's {orch-namespace} as a file in single-workspace or multiple-workspace containers of {prod-short}.
+In {prod-short}, you can mount the {platforms-name} secret from your {orch-namespace} as a file in single-workspace or multiple-workspace containers.
```
What was wrong with the orig. version? In fact, I think it was more descriptive. The new formulation only vaguely suggests that something *can* be done.

max-cx replied:
I already answered this question in the parallel comment in modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc .

rkratky commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -7,23 +7,57 @@
-* `app.kubernetes.io/part-of: che.eclipse.org` 
-* `app.kubernetes.io/component: workspace-secret`:
+. Create a new secret in the {platforms-namespace} where a new workspace is deployed.
+. Match the secret's `labels` to the set of labels configured in the `che.workspace.provision.secret.labels` property of {prod-short}.
```
Don't anthropomorphize by using the possessive. IBM SG, pg. 26.
```suggestion
. Match the `labels` of the secret to the set of labels configured in the `che.workspace.provision.secret.labels` property of {prod-short}.
```

max-cx replied:
@rkratky Our IBM SG pagination is different. May I ask which IBM SG version are you using? I'm using ibm_style_062121_Q2_2021, page 59:

Use the 's construction for the possessive form when it provides the clearest and most direct wording.
✗ Less direct: If the object is a child node, set the isLeaf property of the object to true.
✓ More direct: If the object is a child node, set the object's isLeaf property to true.
Use an alternative construction for the possessive form when it provides the clearest and most direct wording.
Alternative constructions include making a possessive noun an adjective or using a possessive noun in a prepositional
phrase.
✗ The computer's keyboard's Alt key stopped working.
✓ The Alt key on the computer keyboard stopped working.

max-cx replied:
@rkratky https://source.redhat.com/groups/public/ccs/ccs_folder/ibm_style_guide

rkratky commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -7,23 +7,57 @@
+. Create a new secret in the {platforms-namespace} where a new workspace is deployed.
+. Match the secret's `labels` to the set of labels configured in the `che.workspace.provision.secret.labels` property of {prod-short}.
+. Define the default behavior for mounting secrets in the `automount-workspace-secret` annotation, which takes a boolean value:
+** `'true'` enables automatic mounting into all workspace containers.
```
Does the value really need to be quotes?

max-cx replied:
@rkratky Have a look, these values are in quotes in four examples (possibly written by the same author): https://www.eclipse.org/che/docs/che-7/end-user-guide/mounting-a-secret-as-a-file-or-an-environment-variable-into-a-workspace-container/ If this doesn't seem right, we can of course double-check with SMEs.

rkratky replied:
I suggest you do check it with an SME because it seems odd. If the quotes don't need to be there, then remove them from the other examples, too.

mshaposhnik replied:
Quote:
`The keys and the values in the map must be strings. In other words, you cannot use numeric, boolean, list or other types for either the keys or the values.`
https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/

rkratky commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -7,23 +7,57 @@
+** `'false'` disables automatic mounting into all workspace containers. (To mount a secret, a user must request it in the devfile component by using the `automountWorkspaceSecrets:true` property.)
+. Set a `mount-path` for the file.
+. Include the `mount-as: file` annotation.
+. Set the filename in the secret's `data` field.
```
```suggestion
. Set the filename in the `data` field of the secret.
```

rkratky commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -7,23 +7,57 @@
+. Set the filename in the secret's `data` field.
 +
-include::example${project-context}-ref_a-kubernetes-secret-as-a-file.adoc[levelofset=+1]
+NOTE: The {orch-name} secret can contain several `data` items. The names of `data` items must match the name of the secret's file mounted into the container.
```
```suggestion
NOTE: The {orch-name} secret can contain several `data` items. The names of the `data` items must match the name of the secret file mounted into the container.
```

rkratky commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -7,23 +7,57 @@
+<3> Enables automatic mounting into all workspace containers.
+<4> Sets the mount path of the file for all workspace containers.
+<5> Mounts the secret as a file.
+<6> Sets the `settings.xml` filename.
```
IBM SG, pg. 329.
```suggestion
<6> Sets the `settings.xml` file name.
```

rkratky commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc:
```diff
@@ -1,23 +1,61 @@
 = Mounting a secret as an environment variable into a workspace container
 
-The following section describes how to mount a {platforms-name} secret from the user’s {orch-namespace} as an environment variable, or variables, into single-workspace or multiple-workspace containers of {prod-short}.
+In {prod-short}, you can mount the {platforms-name} secret from your {orch-namespace} as an environment variable (or variables) in single-workspace or multiple-workspace containers.
```
What was wrong with the orig. version? In fact, I think it was more descriptive. The new formulation only vaguely suggests that something *can* be done. I'd just use "*This* section".

max-cx replied:
@rkratky IBM GS, p 22: "Avoid self-referential text whenever possible, such as "This topic is about" or "This section describes". Instead, focus on the purpose or value of the content, and include words that users are likely to search for."
This is a tricky topic, and I haven't come up with a standard solution for myself. Also, in a way it's something of a fluffy truism - this section is about what the content of this section is. The trick is how to phrase the introductory sentence: in conceptual modules this isn't a problem, but in procedural modules is where we start getting "You can ...".
By the way, I've already encountered complications with "This section describes": when the section gains siblings or subsections (These sections? The following sections? The sections in a different .adoc file?), the sentence becomes outdated and is less likely to be updated.
Also, if I skip this guidance, then it will be raised during reviews.
@themr0c Any thoughts?

max-cx replied:
@rkratky and @themr0c :) Speaking of fluff, what else I noticed about introductory sentences to procedures is that they often contain the reused phrase from the heading, and so the same phrase appears on two adjacent lines. I don't have a solution to offer, but it does look like fluff.

rkratky replied:
> IBM GS, p 22: "Avoid self-referential text whenever possible, such as "This topic is about" or "This section describes". Instead, focus on the purpose or value of the content, and include words that users are likely to search for."

Good point.

> in procedural modules is where we start getting "You can ...".

_You can_ in general is superfluous -- of course you can. The question is whether you are supposed to do it, or not. Our docs (not just Che) overuse _you can_.

We should describe what the procedure accomplishes. In this case, something like:

_{prod-short} allows mounting a {platforms-name} secret from the user’s {orch-namespace} as an environment variable, or variables, into single-workspace or multiple-workspace containers._

> I've already encountered complications with "This section describes": when the section gains siblings or subsections (These sections? The following sections? The sections in a different .adoc file?), the sentence becomes outdated and is less likely to be updated.

Never refer to _other_ sections. Always consider that all modules (units of content) need to make sense on their own, and the reader may not follow the doc in a linear fasion. Forf this, it's useful to use the _"Every page is page one" check. IOW, always ask yourself whether a module/section would make sense to the reader if they started reading right there.

> ntroductory sentences to procedures is that they often contain the reused phrase from the heading, and so the same phrase appears on two adjacent lines

Yeah. Ideally, the intro/context paragraph should have a bit broader description. But in a pinch, a repetition is better than not intro sentence, I'd say.

max-cx replied:
@rkratky Thank you, Robert. This is very helpful!

rkratky commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc:
```diff
@@ -1,23 +1,61 @@
-* `app.kubernetes.io/component: workspace-secret`:
+. Create a new secret in the {platforms-namespace} where a new workspace is deployed.
+. Define the default behavior for mounting secrets in the `automount-workspace-secret` annotation, which takes a boolean value:
+** `'true'` enables automatic mounting into all workspace containers.
```
Quotes needed?

max-cx replied:
I already replied in response to another similar comment here.

rkratky commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc:
```diff
@@ -1,23 +1,61 @@
+. Define the default behavior for mounting secrets in the `automount-workspace-secret` annotation, which takes a boolean value:
+** `'true'` enables automatic mounting into all workspace containers.
+** `'false'` disables automatic mounting into all workspace containers. (To mount a secret, a user must request it in the devfile component by using the `automountWorkspaceSecrets:true` property.)
+. Set the name of the secret's environment variable in the `env-name` annotation.
```
```suggestion
. Set the name of the secret environment variable in the `env-name` annotation.
```

max-cx replied:
@rkratky I understand your point on why we need to drop the apostrophe, but I only added the apostrophe to separate `the secret` and the `environment variable`.
Otherwise, it can be misunderstood as an `environment variable` that is `secret` because `secret` is also an adjective (in addition to being a noun as intended above).
As an alternative, we may put something wordier here:
`". Set the name of the environment variable of the secret in the `env-name` annotation."`

rkratky commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc:
```diff
@@ -1,23 +1,61 @@
+** `'false'` disables automatic mounting into all workspace containers. (To mount a secret, a user must request it in the devfile component by using the `automountWorkspaceSecrets:true` property.)
+. Set the name of the secret's environment variable in the `env-name` annotation.
+. Include the `mount-as: env` annotation.
+. Match the secret's `labels` to the set of labels configured in the `che.workspace.provision.secret.labels` property of {prod-short}.
```
```suggestion
. Match the `labels` of the secret to the set of labels configured in the `che.workspace.provision.secret.labels` property of {prod-short}.
```

rkratky commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc:
```diff
@@ -1,23 +1,61 @@
+. Set the name of the secret's environment variable in the `env-name` annotation.
+. Include the `mount-as: env` annotation.
+. Match the secret's `labels` to the set of labels configured in the `che.workspace.provision.secret.labels` property of {prod-short}.
+. Set the secret's key-value pair in the secret's `data` field.
```
```suggestion
. Set the key-value pair of the secret in the `data` field of the secret.
```

rkratky commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc:
```diff
@@ -1,23 +1,61 @@
+. Include the `mount-as: env` annotation.
+. Match the secret's `labels` to the set of labels configured in the `che.workspace.provision.secret.labels` property of {prod-short}.
+. Set the secret's key-value pair in the secret's `data` field.
+. If the secret must have multiple key-value pairs in the `data` field, set a different environment variable name with the `<__key__>_env-name` annotation for each key-value pair.
```
Avoid 'if' unless necessary. Describe what to do. Don't introduce the need for weighing decisions.
```suggestion
. To set multiple key-value pairs in the `data` field of the secret, use a different environment variable name with the `__<key>___env-name` annotation format for each key-value pair.
```

max-cx replied:
@rkratky I considered this sentence a conditional because the user does not have to complete this step unless the condition is met: "If the condition is met, then [you must] do something." If we rewrite this as "To do something, do something," then, in addition to making it ambiguous ("To do something" = the underlying reason or purpose or condition?), we lose the conditional and expect the user to do something regardless of whether the condition is met or not. Here is my counter-proposal:

`. If you set multiple key-value pairs in the `data` field of the secret, then use a different environment variable name with the `__<key>___env-name` annotation format for each key-value pair.`

rkratky commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc:
```diff
@@ -1,23 +1,61 @@
+. If the secret must have multiple key-value pairs in the `data` field, set a different environment variable name with the `<__key__>_env-name` annotation for each key-value pair.
 +
-include::example${project-context}-ref_a-kubernetes-secret-as-an-variable.adoc[levelofset=+1]
+NOTE: The maximum length of an annotation name in a {orch-name} secret is 63 characters, of which 9 characters are reserved for a prefix that ends with `/`. This acts as a restriction for the maximum length of the key that can be used for the secret.
```
```suggestion
NOTE: The maximum length of an annotation name in a {orch-name} secret is 63 characters, of which 9 characters are reserved for a prefix that ends with `/`. This restrics the maximum length of the key that can be used for the secret.
```

rkratky commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc:
```diff
@@ -1,23 +1,61 @@
+----
+<1> Enables automatic mounting into all workspace containers.
+<2> Mounts the secret as a variable. 
+<3> Provisions the `__<FOO_ENV>__` environment variable into all workspace containers using the value of `__<key_1>__`.
```
Don't use *foo*, *bar*, or *fubar*/*foobar*. It's slangy.
```suggestion
<3> Provisions the `__<KEY_ENV>__` environment variable into all workspace containers using the value of `__<key_1>__`.
```

rkratky commented on modules/end-user-guide/partials/proc_overriding-a-secrets-mount-path-for-a-workspace-component.adoc:
```diff
@@ -0,0 +1,42 @@
+// mounting-a-secret-as-a-file-or-an-environment-variable-into-a-workspace-container
+
+[id="overriding-a-secrets-mount-path-for-a-workspace-component_{context}"]
+= Overriding a secret's mount path for a workspace component
```
This shouldn't be a possessive, but given that it's reflected in the file name, I leave it up to you when to fix it (doesn't need to be now).

rkratky commented on modules/end-user-guide/partials/proc_overriding-a-secrets-mount-path-for-a-workspace-component.adoc:
```diff
@@ -0,0 +1,42 @@
+endif::[]
+OpenShift {ocp3-ver}, secrets mounted as a file override volume mounts defined in the devfile.
+
+You can override a secret's mount path for a workspace component by editing the devfile.
```
```suggestion
You can override the mount path of a secret for a workspace component by editing the devfile.
```

max-cx replied:
@rkratky I only added the possessive here to avoid the misunderstanding that the secret is only for one workspace component ("a secret for a workspace component").

rkratky commented on modules/end-user-guide/partials/proc_overriding-a-secrets-mount-path-for-a-workspace-component.adoc:
```diff
@@ -0,0 +1,42 @@
+. Use the name of the secret as the name of the volume.
+. Set the new mount path for the secret as `containerPath`.
++
+.Example: Overriding a secret's mount path for a workspace component
```
```suggestion
.Example: Overriding the mount path of a secret for a workspace component
```

max-cx replied:
I'm changing it to *Overriding the mount path of the secret for a workspace component*.

rkratky reviewed (changes requested):
Comments & suggestions added.

max-cx committed:
Update modules/end-user-guide/partials/assembly_mounting-secrets-as-file-or-env-variable-in-workspace-container.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/end-user-guide/partials/assembly_mounting-secrets-as-file-or-env-variable-in-workspace-container.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

max-cx committed:
Update modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc

Co-authored-by: Robert Krátký <rkratky@redhat.com>

themr0c committed:
Merge branch 'master' into 3166

themr0c committed:
Merge branch 'master' into 3166

themr0c committed:
Merge branch 'master' into 3166

max-cx committed:
fix typo

max-cx committed:
Merge branch 'master' into 3166

max-cx committed:
[3166] fix sentences

 Changes to be committed:
	modified:   modules/end-user-guide/partials/proc_mounting-a-secret-as-an-environment-variable-into-a-workspace-container.adoc

max-cx committed:
[3166] fix: change filename, heading, instruction text

Before:
[id="overriding-a-secrets-mount-path-for-a-workspace-component_{context}"]
= Overriding a secret's mount path for a workspace component
After:
[id="overriding-the-mount-path-of-the-secret-for-a-workspace-component_{context}"]
= Overriding the mount path of the secret for a workspace component

Before:
'a secret's mount path for a workspace component'
After:
'the mount path of the secret for a workspace component'

 Changes to be committed:
	modified:   modules/end-user-guide/partials/assembly_mounting-secrets-as-file-or-env-variable-in-workspace-container.adoc
	modified:   modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc
	renamed:    modules/end-user-guide/partials/proc_overriding-a-secrets-mount-path-for-a-workspace-component.adoc -> modules/end-user-guide/partials/proc_overriding-the-mount-path-of-the-secret-for-a-workspace-component.adoc

Bug: https://issues.redhat.com/browse/RHDEVDOCS-3166

max-cx committed:
[3166] fix: change filename, heading, instruction text

Before:
[id="overriding-a-secrets-mount-path-for-a-workspace-component_{context}"]
= Overriding a secret's mount path for a workspace component
After:
[id="overriding-the-mount-path-of-the-secret-for-a-workspace-component_{context}"]
= Overriding the mount path of the secret for a workspace component

Before:
'a secret's mount path for a workspace component'
After:
'the mount path of the secret for a workspace component'

Changes to be committed:
	renamed:    modules/end-user-guide/partials/proc_overriding-a-secrets-mount-path-for-a-workspace-component.adoc -> modules/end-user-guide/partials/proc_overriding-the-mount-path-of-the-secret-for-a-workspace-component.adoc

Bug: https://issues.redhat.com/browse/RHDEVDOCS-3166

max-cx committed:
fix: minor edits

max-cx committed:
Merge branch 'master' into 3166

max-cx committed:
fix: 'FOO_ENV' to 'VAR_1; 'OTHER_ENV' to 'VAR_2'

themr0c committed:
Merge branch 'master' into 3166

max-cx committed:
Merge branch 'master' into 3166

max-cx commented on modules/end-user-guide/partials/assembly_mounting-secrets-as-file-or-env-variable-in-workspace-container.adoc:
```diff
@@ -11,25 +11,27 @@ Secrets are {platforms-name} objects that store sensitive data such as user name
 +
 NOTE: SSH key pairs can also be mounted as a file, but this format is primarily aimed at the settings of the Maven configuration.
 
 The mounting process uses the standard {platforms-name} mounting mechanism, but it requires additional annotations and labeling for a proper bound of a secret with the required {prod-short} workspace container.
```
```suggestion
The mounting process uses the standard {platforms-name} mounting mechanism, but it requires additional annotations and labeling to properly bind a secret with the required {prod-short} workspace container.
```

max-cx committed:
Update modules/end-user-guide/partials/assembly_mounting-secrets-as-file-or-env-variable-in-workspace-container.adoc

max-cx committed:
Merge branch 'master' into 3166

max-cx committed:
Merge branch 'master' into 3166

MichalMaler committed:
Merge branch 'master' into 3166

max-cx committed:
Merge branch 'master' into 3166

themr0c commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -7,23 +7,57 @@
-* `app.kubernetes.io/part-of: che.eclipse.org` 
-* `app.kubernetes.io/component: workspace-secret`:
+. Create a new secret in the {platforms-namespace} where a new workspace is deployed.
+. Match the `labels` of the secret to the set of labels configured in the `che.workspace.provision.secret.labels` property of {prod-short}.
```
The information about the default value is lost.

themr0c commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -7,23 +7,57 @@
+ifeval::["{project-context}" == "che"]
+* For more information about creating secrets, see link:https://kubernetes.io/docs/concepts/configuration/secret/#creating-a-secret[Kubernetes Documentation].
+endif::[]
+* xref:mounting-a-secret-as-a-file-or-an-environment-variable-into-a-workspace-container.adoc#the-use-of-annotations-in-the-process-of-mounting-a-secret-into-a-workspace-container_{context}[]
```
Deep link need a title in `[]`.

themr0c commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -7,23 +7,57 @@
+<4> Sets the mount path of the file for all workspace containers.
+<5> Mounts the secret as a file.
+<6> Sets the `settings.xml` file name.
+
```
Missing a verification step.

themr0c commented on modules/end-user-guide/partials/proc_mounting-a-secret-as-a-file-into-a-workspace-container.adoc:
```diff
@@ -7,23 +7,57 @@
+** `'false'` disables automatic mounting into all workspace containers. (To mount a secret, a user must request it in the devfile component by using the `automountWorkspaceSecrets:true` property.)
+. Set a `mount-path` for the file.
+. Include the `mount-as: file` annotation.
+. Set the filename in the `data` field of the secret.
```
Missing a step explaining how to encode the secret in base64. See example in https://www.eclipse.org/che/docs/che-7/end-user-guide/provisioning-the-jetbrains-offline-activation-code/.

max-cx committed:
Merge remote-tracking branch 'upstream/master' into 3166

max-cx committed:
draft

max-cx committed:
draft

themr0c committed:
Merge branch 'master' into 3166
